chain/client/src/stateless_validation/partial_witness/partial_witness_tracker.rs

```diff
@@ -242,30 +252,27 @@ impl PartialEncodedStateWitnessTracker {
         let num_parts = self.get_num_parts(&partial_witness)?;
         let rs = self.rs_map.entry(num_parts);
         let new_entry = CacheEntry::new(rs);
-        if let Some((evicted_chunk_hash, evicted_entry)) = self.parts_cache.push(key, new_entry) {
+        if let Some((evicted_key, evicted_entry)) = self.parts_cache.push(key, new_entry) {
             // Record the ratio of parts received to parts required for the evicted entry.
             // Note that this includes the parts received after decoding the state witness.
             let parts_received_ratio =
```

Comment:
Tiny issue here. We were originally relying on the eventual eviction of the cache entry to fill in these metrics. This includes receiving and keeping track of parts AFTER we've received enough parts to keep track of ratios of parts received to total parts.

With the new design, we pop the entry as soon as we have enough parts. I guess we can get rid of `PARTIAL_WITNESS_PARTS_RECEIVED_RATIO` and  `PARTIAL_WITNESS_TOTAL_TIME` metrics and retain the warning, maybe even convert it to an error?

Reply:
ah indeed, thanks for catching that!
done in 6fc2fa274, decided to still log it as a warning, since it doesn't seem critical enough to be an error